aesara/tensor/random/type.py

```diff
@@ -28,7 +28,14 @@ def filter(cls, data, strict=False, allow_downcast=None):
 
     @staticmethod
     def is_valid_value(a):
-        return isinstance(a, np.random.RandomState)
+        if isinstance(a, np.random.RandomState):
```

Comment:
But if the our goal is only limited to storage of the `RNG` array and not reconverting back into `RandomState`, This approach is also worth considering i.e simply make `RandomStateType` compatible with arrays rather than sub-classing it. Though we can also reconvert those arrays back to `RandomState` during filtering.

Reply:
Yes, that's a viable approach as well, especially since `Type.filter` has a `strict` keyword argument that we can use to justify this kind of overloading.

Reply:
If we are to verify all the other parameters that you mentioned, we could pass on the entire information provided by the newer API instead of just passing the state key. That way, the identification of a `jax_typified` `RandomState` would also be easier than just being a array-like object with a specified shape.